src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -209,28 +199,34 @@ internal static bool IsRuntimeAssembly(string assemblyPath)
 
         internal static bool IsRuntimeAssembly(AssemblyName assemblyName)
         {
-            var publicKeyToken = assemblyName.GetPublicKeyToken();
-            var lowerInvariant = BitConverter.ToString(publicKeyToken).Replace("-", string.Empty).ToLowerInvariant();
+            var keyToken = assemblyName.GetPublicKeyToken();
+            if (keyToken.Length == 0)
+            {
+                return false;
+            }
+
+            var publicKeyToken = BitConverter.ToInt64(keyToken, 0);
+            
             //System
-            if (lowerInvariant == "b77a5c561934e089")
+            if (publicKeyToken == -8511746012302509385)
```

Comment:
I mean you are right it is not a drastic improvement. Maybe not worth it from the overall picture. But on the other hand it wouldn't hurt either